packages/esbuild/esbuild_repositories.bzl

```diff
@@ -0,0 +1,108 @@
+"""
+Helper macro for fetching esbuild versions for internal tests and examples in rules_nodejs
+"""
+
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@build_bazel_rules_nodejs//internal/common:check_version.bzl", "parse_version")
+
+_VERSION = "0.12.1"
+_MIN_VERSION = "0.12.0"
+_MAX_VERSION = "0.13.0"
+
+DARWIN_AMD64 = "darwin_amd64"
+LINUX_AMD64 = "linux_amd64"
+WINDOWS_AMD64 = "windows_amd64"
+
+ESBUILD_SHAS = dict({
+    DARWIN_AMD64: "efb34692bfa34db61139eb8e46cd6cf767a42048f41c8108267279aaf58a948f",
+    LINUX_AMD64: "de8409b90ec3c018ffd899b49ed5fc462c61b8c702ea0f9da013e0e1cd71549a",
+    WINDOWS_AMD64: "10439647b11c7fd1d9647fd98d022fe2188b4877d2d0b4acbe857f4e764b17a9",
+})
+
+def _maybe(repo_rule, name, **kwargs):
+    if name not in native.existing_rules():
+        repo_rule(name = name, **kwargs)
+
+def _esbuild_toolchain_repository_impl(repository_ctx):
+    content = "SUPPORTED_PLATFORMS = %s" % str(repository_ctx.attr.platforms)
+    repository_ctx.file("index.bzl", content = content)
+    repository_ctx.file("BUILD.bazel")
+
+_esbuild_toolchain_repository = repository_rule(
+    implementation = _esbuild_toolchain_repository_impl,
+    attrs = {
+        "platforms": attr.string_list(
+            mandatory = True,
+        ),
+    },
+)
+
+def esbuild_repositories(name = "esbuild_repositories", version = _VERSION, shas = ESBUILD_SHAS):
+    """Helper for fetching and setting up the esbuild versions and toolchains
+
+    Args:
+        name: currently unused
```

Comment:
if the user passes some other name, is that illegal since you rely on this name elsewhere? should we check?

Reply:
No, as the doc states the name is currently unused, it's not relied on anywhere. There is a buildifier warning when there is no name attr.